packages/esm-appointments-app/src/patient-queue/visit-form/base-visit-type.component.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import debounce from 'lodash-es/debounce';
+import isEmpty from 'lodash-es/isEmpty';
+import { Layer, Search, RadioButtonGroup, RadioButton, Tile } from '@carbon/react';
+import { useLayoutType, usePagination } from '@openmrs/esm-framework';
+import styles from './base-visit-type.scss';
+import EmptyState from '../../empty-state/empty-state.component';
+
+interface BaseVisitTypeProps {
+  onChange: (event) => void;
+  patientUuid: string;
+  visitTypes;
+}
+
+const BaseVisitType: React.FC<BaseVisitTypeProps> = ({ onChange, visitTypes }) => {
+  const { t } = useTranslation();
+  const isTablet = useLayoutType() === 'tablet';
+  const [searchTerm, setSearchTerm] = useState<string>('');
+
+  const searchResults = useMemo(() => {
+    if (!isEmpty(searchTerm)) {
+      return visitTypes.filter((visitType) => visitType.display.toLowerCase().search(searchTerm.toLowerCase()) !== -1);
+    } else {
+      return visitTypes;
+    }
+  }, [searchTerm, visitTypes]);
+
+  const handleSearch = React.useMemo(() => debounce((searchTerm) => setSearchTerm(searchTerm), 300), []);
+
+  const { results } = usePagination<{ uuid: string; display: string; name: string }>(searchResults, 5);
+
+  return (
+    <div className={`${styles.visitTypeOverviewWrapper} ${isTablet ? styles.tablet : styles.desktop}`}>
+      {results.length ? (
```

Comment:
Nested ternaries are pretty hard to read. Could the EmptyState be broken out into its on return statement above? `if (!results.length) return <EmptyState ... /> ;`